### PR TITLE
ssl_session_reuse optimization to check if diags is on

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

File: plugins/experimental/ssl_session_reuse/src/publish.cc [hunk 1]
```diff
@@ -117,7 +117,11 @@ RedisPublisher::is_good()
 ::redisContext *
 RedisPublisher::setup_connection(const RedisEndpoint &re)
 {
-  auto my_id = (unsigned long long)::pthread_self();
+  uint64_t my_id = 0;
+  if (TSIsDebugTagSet(PLUGIN)) {
+    my_id = (uint64_t)pthread_self();
+  }
+
   TSDebug(PLUGIN, "RedisPublisher::setup_connection: Called by threadId: %llx", my_id);
```
Comment (maskit) [inline plugins/experimental/ssl_session_reuse/src/publish.cc]: Shouldn't we put this line into the if block?

Comment (maskit) [reply plugins/experimental/ssl_session_reuse/src/publish.cc]: You do it for some of other places. But not here?

Comment (bryancall) [reply plugins/experimental/ssl_session_reuse/src/publish.cc]: I can do it for this one easily.  However, there are more TSDebug() function calls in this function.

Comment (maskit) [reply plugins/experimental/ssl_session_reuse/src/publish.cc]: OK.

**Comment by maskit** [comment]
For `uint64_t` we need to use `PRIx64` instead of `%llx`.

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**